assimilation_code/programs/preprocess/preprocess.f90

```diff
@@ -2,57 +2,112 @@
 ! by UCAR, "as is", without charge, subject to all terms of use at
 ! http://www.image.ucar.edu/DAReS/DART/DART_download
 !
-! $Id$
 
-!> Takes a list of observation type module path names. These modules contain
-!> multiple fragments of standard F90 that may be required to implement forward
-!> observation operators for DART. The sections are retrieved from the files
-!> by this program and inserted into the appropriate blanks in the
-!> DEFAULT_obs_def_mod.F90 and DEFAULT_obs_kind_mod.F90 templates. 
+!> Define the observation types and obs/state quantities understood by DART.
+!>
+!> Takes a list of observation type module path names and physical quantity
+!> module path names and creates two output files that are compiled into
+!> the rest of the DART executables.  They define the quantities and observation
+!> types that are recognized and processed in any way. 
+!>
+!> The output module for observation types can contain multiple fragments of 
+!> standard F90 that may be required to implement forward observation operators 
+!> for DART. The sections are retrieved from the files by this program and 
+!> inserted into the appropriate blanks in the DEFAULT_obs_def_mod.F90 and 
+!> DEFAULT_obs_kind_mod.F90 templates. 
+!>
 !> The final obs_def_mod.f90 and obs_kind_mod.f90 that are created contain
 !> the default code plus all the code required from the selected observation
-!> type modules. Preprocess also inserts the required identifier and string
-!> for the corresponding observation kinds (and only those kinds).
+!> type modules. Preprocess also inserts the required identifiers and strings
+!> for the corresponding observation quantities.
+!>
+!> @todo FIXME: the module name is set by whatever is in the DEFAULT_xxx file
+!> which is still obs_kind_mod instead of _qty_.  also i have changed the default
+!> filenames here to be _qty_(HK?) but in reality all the pathnames files for the
+!> build system expect _kind_ and the names are always specified in our input.nml
+!> namelist files, so we aren't depending on the defaults. (maybe we should?)
+!> and finally, this code generates module 'use' statements, and those, 
+!> to keep things working, are still outputing "use obs_kind_mod" instead 
+!> of "use obs_qty_mod".   this should change but i'm not sure when.
+!>
+!> @todo FIXME: this code now creates an obs_def_mod.f90 that contains
```

Comment:
i think this comment is obsolete and can be removed.  there is no way for preprocess to really know what QTYs are needed by the model_mods or any other code being compiled into an executable (e.g. utilitity programs, obs_loop programs, etc).   adding them to the quantity modules avoids this issue.